Sources/Purchasing/StoreKit2/StoreKit2TransactionListener.swift

```diff
@@ -37,7 +37,12 @@ class StoreKit2TransactionListener {
     func listenForTransactions() {
         self.taskHandle?.cancel()
         self.taskHandle = Task { [weak self] in
-            for await result in StoreKit.Transaction.updates {
+            // Transaction.updates only sends updates from transactions made on different devices,
+            // so on observer mode we need to get `all` instead.
+
+            // TODO: ability to access observer mode setting from here
+            let transactionsStream = observerModeEnabled ? StoreKit.Transaction.all : StoreKit.Transaction.updates
```

Comment:
@NachoSoto I can wrap this up tonight if you don't have the time, basically we need to pass in system info or the setting here